### PR TITLE
Adding KLE image for layout support clarification

### PR DESCRIPTION
**Pull request by lukelex**
Adding an image showcasing the supported layouts for easier understanding of new users.

This should help on https://github.com/qmk/qmk_firmware/pull/7726

## Description

* Added KLE layout image to Readme.

## Types of Changes

- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).

### CODE REVIEW

**Comment by noroadsleft** [comment]
Also as an aside, the colors on your mock-ups for the split shifts have the wrong colors (split Right Shift mock-up has left Shift's color and vice versa). :wink:

**Comment by lukelex** [comment]
What would be the proper solution for this then?

**Comment by noroadsleft** [comment]
Someone would need to add the layout macros to `lex60.h` and their corresponding layout data to `info.json`.

File: keyboards/gami_studio/lex60/readme.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 A 60% with split space bars made by Gami Studio.
 
+![Lex60 Layout support](https://imgur.com/a/sVP3s0o)
```
Comment (noroadsleft) [inline keyboards/gami_studio/lex60/readme.md]: This needs to be a direct image URL for GitHub to render the image on the page.

```suggestion
![Lex60 Layout support](https://i.imgur.com/oehz1Zm.png)
```

**Comment by noroadsleft** [comment]
> Someone would need to add the layout macros to `lex60.h` and their corresponding layout data to `info.json`.

The [GH60 Satan codebase](https://github.com/qmk/qmk_firmware/tree/master/keyboards/gh60/satan) is a good example of supporting multiple physical layouts on the same hardware. `satan.h` has the arguments and the matrix positions for each physical layout currently supported, and then `info.json` has the physical layout data (where each key is and it's size) for each layout macro.

I could actually add it in to my PR – it's actually my specialty around here – but I'd need you to check the layout or test them if possible, as I don't have the hardware.

**Comment by lukelex** [comment]
@noroadsleft @zvecr I've added a couple of supported layouts as both macros and layouts. I've also tested them all on the physical boards. How should I go about editing the `info.json`?

**Comment by noroadsleft** [comment]
> I've added a couple of supported layouts as both macros and layouts. I've also tested them all on the physical boards. How should I go about editing the `info.json`?

## Create KLE Raw Data

Take the arguments from https://github.com/qmk/qmk_firmware/blob/b956e9caf626ea9dfd69125c2db327f4dda858e5/keyboards/gami_studio/lex60/lex60.h#L8-L14

... and turn it into a KLE Raw Data block:

```
["k00", "k01", "k02", "k03", "k04", "k05", "k06", "k07", "k08", "k09", "k0a", "k0b", "k0c", "k0d", "k0e"], 
["k10", "k11", "k12", "k13", "k14", "k15", "k16", "k17", "k18", "k19", "k1a", "k1b", "k1c", "k1d"], 
["k20", "k21", "k22", "k23", "k24", "k25", "k26", "k27", "k28", "k29", "k2a", "k2b", "k2c", "k2d"], 
["k30", "k31", "k32", "k33", "k34", "k35", "k36", "k37", "k38", "k39", "k3a", "k3b", "k3c", "k3d"], 
["k40", "k41", "k42", "k45", "k46", "k4a", "k4b", "k4c", "k4d"]
```

Then paste that block into the Raw Data tab on Keyboard Layout Editor:

image

## Size the keys to match the hardware

Resize the keys in KLE to how they are on the assembled board:

image

## Generate the `info.json` file

Then take the resultant Raw Data and copy-paste it into https://qmk.fm/converter/, and hit the `Convert` button.

Copy the output to a new file and save that file as `keyboards/gami_studio/lex60/info.json`.

To add additional layouts, repeat the above steps, pasting the Converter Output into the `info.json` file, such that the completed file has the JSON tree:

- keyboard_name
- url
- maintainer
- width
- height
- layouts
  - LAYOUT_all
    - layout
  - LAYOUT_uhhkb
    - layout
  - LAYOUT_hhkb
    - layout
  - LAYOUT_iso
    - layout

Make sure the key names under the `layouts` tree are exact matches to what you named the macros in `lex60.h`.

The `LAYOUT_iso` output from the Converter will be wrong when you feed it that layout; it'll put `k2d` between `k1c` key and `k20`. Take the `k2d` JSON object, and move it to after the object for `k2c`. 

Or don't, and I'll give you a suggestion to correct it on PR review.

File: keyboards/gami_studio/lex60/lex60.h [hunk 2]
```diff
@@ -19,3 +19,49 @@
...
+) \
+{ \
+    {k00, k01, k02, k03, k04, k05, k06, k07, k08, k09, k0a, k0b, k0c, k0d, k0e}, \
+    {k10, k11, k12, k13, k14, k15, k16, k17, k18, k19, k1a, k1b, k1c, k1d, XXX}, \
+    {k20, k21, k22, k23, k24, k25, k26, k27, k28, k29, k2a, k2b, XXX, k2d, XXX}, \
+    {k30, XXX, k32, k33, k34, k35, k36, k37, k38, k39, k3a, k3b, k3c, XXX, XXX}, \
+    {XXX, k41, k42, XXX, XXX, k45, k46, XXX, XXX, k49, k4a, k4b, XXX, XXX, XXX}  \
+}
+
+#define LAYOUT_hhkb( \
```
Comment (noroadsleft) [inline keyboards/gami_studio/lex60/lex60.h]: HHKB layout is one of our Community Layouts.

[LAYOUT_60_hhkb](https://github.com/qmk/qmk_firmware/tree/abd36de5adbb484695d3500ad790df557d4f5419/layouts/default/60_hhkb)

```suggestion
#define LAYOUT_60_hhkb( \
```

File: keyboards/gami_studio/lex60/lex60.h [hunk 3]
```diff
@@ -19,3 +19,49 @@
...
+) \
+{ \
+    {k00, k01, k02, k03, k04, k05, k06, k07, k08, k09, k0a, k0b, k0c, k0d, k0e}, \
+    {k10, k11, k12, k13, k14, k15, k16, k17, k18, k19, k1a, k1b, k1c, k1d, XXX}, \
+    {k20, k21, k22, k23, k24, k25, k26, k27, k28, k29, k2a, k2b, XXX, k2d, XXX}, \
+    {k30, XXX, k32, k33, k34, k35, k36, k37, k38, k39, k3a, k3b, k3c, k3d, XXX}, \
+    {XXX, k41, k42, XXX, XXX, XXX, k46, XXX, XXX, XXX, XXX, k4b, k4c, XXX, XXX}  \
+}
+
+#define LAYOUT_iso( \
```
Comment (noroadsleft) [inline keyboards/gami_studio/lex60/lex60.h]: So is a standard 60% ISO.

[LAYOUT_60_iso](https://github.com/qmk/qmk_firmware/tree/abd36de5adbb484695d3500ad790df557d4f5419/layouts/default/60_iso)

```suggestion
#define LAYOUT_60_iso( \
```

File: keyboards/gami_studio/lex60/keymaps/iso/keymap.c [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+// entirely and just use numbers.
+enum layer_names {
+  _BASE,
+  _GAMING,
+  _FUNCTION,
+  _FIRMWARE
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_BASE] = LAYOUT_iso(
```
Comment (noroadsleft) [inline keyboards/gami_studio/lex60/keymaps/iso/keymap.c]:
```suggestion
  [_BASE] = LAYOUT_60_iso(
```

File: keyboards/gami_studio/lex60/keymaps/iso/keymap.c [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      KC_GESC,              KC_1,    KC_2,    KC_3,     KC_4,    KC_5,    KC_6,    KC_7,          KC_8,    KC_9,     KC_0,    KC_MINS, KC_EQL,  KC_BSPC,
+      KC_TAB,               KC_Q,    KC_W,    KC_E,     KC_R,    KC_T,    KC_Y,    KC_U,          KC_I,    KC_O,     KC_P,    KC_LBRC, KC_RBRC,
+      MT(MOD_LCTL, KC_ESC), KC_A,    KC_S,    KC_D,     KC_F,    KC_G,    KC_H,    KC_J,          KC_K,    KC_L,     KC_SCLN, KC_QUOT, KC_BSLS, KC_ENT,
+      KC_LSFT,              KC_NUBS, KC_Z,    KC_X,     KC_C,    KC_V,    KC_B,    KC_N,          KC_M,    KC_COMM,  KC_DOT,  KC_SLSH, KC_RSFT,
+      KC_LCTL,              KC_LALT, KC_LGUI,                             LT(2, KC_SPC),                   KC_RGUI,  KC_RALT, KC_APP,  MO(3)),
+
+  [_GAMING] = LAYOUT_iso(
```
Comment (noroadsleft) [inline keyboards/gami_studio/lex60/keymaps/iso/keymap.c]:
```suggestion
  [_GAMING] = LAYOUT_60_iso(
```

File: keyboards/gami_studio/lex60/keymaps/iso/keymap.c [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      KC_GESC,              KC_1,    KC_2,    KC_3,     KC_4,    KC_5,    KC_6,    KC_7,          KC_8,    KC_9,     KC_0,    KC_MINS, KC_EQL,  KC_BSPC,
+      KC_TAB,               KC_Q,    KC_W,    KC_E,     KC_R,    KC_T,    KC_Y,    KC_U,          KC_I,    KC_O,     KC_P,    KC_LBRC, KC_RBRC,
+      KC_LCTL,              KC_A,    KC_S,    KC_D,     KC_F,    KC_G,    KC_H,    KC_J,          KC_K,    KC_L,     KC_SCLN, KC_QUOT, KC_BSLS, KC_ENT,
+      KC_LSFT,              KC_NUBS, KC_Z,    KC_X,     KC_C,    KC_V,    KC_B,    KC_N,          KC_M,    KC_COMM,  KC_DOT,  KC_SLSH, KC_RSFT,
+      KC_LCTL,              KC_LALT, KC_LGUI,                             KC_SPC,                          KC_RGUI,  KC_RALT, KC_APP,  MO(3)),
+
+  [_FUNCTION] = LAYOUT_iso(
```
Comment (noroadsleft) [inline keyboards/gami_studio/lex60/keymaps/iso/keymap.c]:
```suggestion
  [_FUNCTION] = LAYOUT_60_iso(
```

File: keyboards/gami_studio/lex60/keymaps/iso/keymap.c [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+      KC_GESC,              KC_F1,   KC_F2,   KC_F3,    KC_F4,   KC_F5,   KC_F6,   KC_F7,         KC_F8,   KC_F9,    KC_F10,  KC_F11,  KC_F12,  KC_DEL,
+      KC_TAB,               XXXXXXX, KC_UP,   XXXXXXX,  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,       XXXXXXX, XXXXXXX,  KC_PSCR, XXXXXXX, XXXXXXX,
+      KC_LCTL,              KC_LEFT, KC_DOWN, KC_RIGHT, XXXXXXX, XXXXXXX, KC_LEFT, KC_DOWN,       KC_UP,   KC_RIGHT, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+      KC_LSFT,              XXXXXXX, XXXXXXX, XXXXXXX,  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,       XXXXXXX, KC_MPRV,  KC_MNXT, KC_MPLY, XXXXXXX,
+      KC_LCTL,              KC_LALT, KC_LGUI,                             LT(2, KC_ENT),                   KC_RGUI,  KC_RALT, KC_APP,  KC_RCTRL),
+
+  [_FIRMWARE] = LAYOUT_iso(
```
Comment (noroadsleft) [inline keyboards/gami_studio/lex60/keymaps/iso/keymap.c]:
```suggestion
  [_FIRMWARE] = LAYOUT_60_iso(
```

File: keyboards/gami_studio/lex60/keymaps/hhkb/keymap.c [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+// entirely and just use numbers.
+enum layer_names {
+  _BASE,
+  _GAMING,
+  _FUNCTION,
+  _FIRMWARE
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_BASE] = LAYOUT_hhkb(
```
Comment (noroadsleft) [inline keyboards/gami_studio/lex60/keymaps/hhkb/keymap.c]:
```suggestion
  [_BASE] = LAYOUT_60_hhkb(
```

File: keyboards/gami_studio/lex60/keymaps/hhkb/keymap.c [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+      KC_GRV,               KC_1,    KC_2,    KC_3,     KC_4,    KC_5,    KC_6,    KC_7,          KC_8,    KC_9,     KC_0,    KC_MINS, KC_EQL,  KC_BSLS,   KC_ESC,
+      KC_TAB,               KC_Q,    KC_W,    KC_E,     KC_R,    KC_T,    KC_Y,    KC_U,          KC_I,    KC_O,     KC_P,    KC_LBRC, KC_RBRC, KC_BSPC,
+      MT(MOD_LCTL, KC_ESC), KC_A,    KC_S,    KC_D,     KC_F,    KC_G,    KC_H,    KC_J,          KC_K,    KC_L,     KC_SCLN, KC_QUOT,          KC_ENT,
+      KC_LSFT,                       KC_Z,    KC_X,     KC_C,    KC_V,    KC_B,    KC_N,          KC_M,    KC_COMM,  KC_DOT,  KC_SLSH, KC_RSFT, MO(3),
+                                     KC_LALT, KC_LGUI,                    LT(2, KC_ENT),          KC_RGUI, KC_RALT),
+
+  [_GAMING] = LAYOUT_hhkb(
```
Comment (noroadsleft) [inline keyboards/gami_studio/lex60/keymaps/hhkb/keymap.c]:
```suggestion
  [_GAMING] = LAYOUT_60_hhkb(
```

File: keyboards/gami_studio/lex60/keymaps/hhkb/keymap.c [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+      KC_GRV,               KC_1,    KC_2,    KC_3,     KC_4,    KC_5,    KC_6,    KC_7,          KC_8,    KC_9,     KC_0,    KC_MINS, KC_EQL,  KC_BSLS,   KC_ESC,
+      KC_TAB,               KC_Q,    KC_W,    KC_E,     KC_R,    KC_T,    KC_Y,    KC_U,          KC_I,    KC_O,     KC_P,    KC_LBRC, KC_RBRC, KC_BSPC,
+      KC_LCTL,              KC_A,    KC_S,    KC_D,     KC_F,    KC_G,    KC_H,    KC_J,          KC_K,    KC_L,     KC_SCLN, KC_QUOT,          KC_ENT,
+      KC_LSFT,                       KC_Z,    KC_X,     KC_C,    KC_V,    KC_B,    KC_N,          KC_M,    KC_COMM,  KC_DOT,  KC_SLSH, KC_RSFT, MO(3),
+                                     KC_LALT, KC_LGUI,                    LT(2, KC_ENT),          KC_RGUI, KC_RALT),
+
+  [_FUNCTION] = LAYOUT_hhkb(
```
Comment (noroadsleft) [inline keyboards/gami_studio/lex60/keymaps/hhkb/keymap.c]:
```suggestion
  [_FUNCTION] = LAYOUT_60_hhkb(
```

File: keyboards/gami_studio/lex60/keymaps/hhkb/keymap.c [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+      KC_GRV,               KC_F1,   KC_F2,   KC_F3,    KC_F4,   KC_F5,   KC_F6,   KC_F7,         KC_F8,   KC_F9,    KC_F10,  KC_F11,  KC_F12,  KC_BSLS,   KC_DEL,
+      KC_TAB,               XXXXXXX, KC_UP,   XXXXXXX,  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,       XXXXXXX, XXXXXXX,  KC_PSCR, XXXXXXX, XXXXXXX, XXXXXXX,
+      KC_LCTL,              KC_LEFT, KC_DOWN, KC_RIGHT, XXXXXXX, XXXXXXX, KC_LEFT, KC_DOWN,       KC_UP,   KC_RIGHT, XXXXXXX, XXXXXXX,          XXXXXXX,
+      KC_LSFT,                       XXXXXXX, XXXXXXX,  XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,       XXXXXXX, KC_MPRV,  KC_MNXT, KC_MPLY, XXXXXXX, MO(3),
+                                     KC_LALT, KC_LGUI,                    LT(2, KC_ENT),          KC_RGUI, KC_RALT),
+
+  [_FIRMWARE] = LAYOUT_hhkb(
```
Comment (noroadsleft) [inline keyboards/gami_studio/lex60/keymaps/hhkb/keymap.c]:
```suggestion
  [_FIRMWARE] = LAYOUT_60_hhkb(
```

File: keyboards/gami_studio/lex60/lex60.h (hunk 2 above)
Comment (lukelex) [reply keyboards/gami_studio/lex60/lex60.h]: Is this gonna work even when there are extra pins when compared to the commonly available layout?

**Pull request closed without merging**